gke-policies/policy/network_policies.rego

```diff
@@ -0,0 +1,51 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# METADATA
+# title: GKE Network Policies engine
+# description: GKE cluster should have Network Policies or Dataplane V2 enabled
+# custom:
+#   group: Security
+package gke.policy.network_policies_engine
+
+default valid = false
+
+valid {
+   count(violation) == 0
+}
+
+
+dataplane_v1_netpol_config {
+  not input.addons_config.network_policy_config.disabled
+}
+
+dataplane_v1_netpol_enabled {
+  not input.network_policy.enabled
+}
+
+dataplane_v1 {
+  dataplane_v1_netpol_config
+  dataplane_v1_netpol_enabled
+}
+
+dataplane_v2 {
+  not input.addons_config.network_policy_config.disabled
+  not input.network_config.datapath_provider == 2
+}
+
+violation[msg] {
```

Comment:
I think we should have at least three violations (logical ORs) for the following:

* Network policy addon is not enabled AND 
* Network Policy config is not enabled AND
* dataplane v2 is not used

OR

* Network policy addon is enabled AND 
* Network Policy config is not enabled AND
* dataplane v2 is not used

OR

* Network policy addon is not enabled AND 
* Network Policy config is enabled AND
* dataplane v2 is not used

In all above cases we don't care about network policy settings if dataplane v2 is used.
I assumed also don't care about Network Policy provider setting